oaknational/Oak-Web-Application
# Feat/firebase proxy

## Description

- This is actually a spike, we probably want to add this to the cloudflare worker list
- Sets `firebase.auth.config.apiHost` and `firebase.auth.config.tokenHost`
  - These options aren't well documented, but internally it's how the endpoints are set, and it works.
- Proxies firebase traffic our own (zero-rated) hostname
- Adds `envName` to config file, so we get clearer messages in console if missing var eg:

![image](https://user-images.githubusercontent.com/12934669/165965675-f6cec074-54ed-4708-8275-c0a653b4ac65.png)


## Issue(s)

Fixes #59 

## How to test

1. Go to https://samara-ijy62bqtu-oak-national-academy.vercel.app/sign-in
2. With network tab open, go through login flow
3. You should see traffic being routed through our own api:
- https://googleapis-identitytoolkit.thenational.academy
- https://securetoken-identitytoolkit.thenational.academy
